cmd/bootstrap/cmd.go

```diff
@@ -0,0 +1,289 @@
+		os.Exit(1)
+	}
+
+	r.Reporter.Infof("VPC created")
+}
+
+// parseParams converts the list of parameter strings into a map and sets default values
+func parseParams(params []string) (map[string]string, map[string]string) {
+	result := map[string]string{
+		"Name":                   "rosa-quickstart-default",
+		"VpcCidr":                "10.0.0.0/16",
+		"Region":                 "us-west-2",
+		"AvailabilityZonesCount": "2",
+	}
+
+	// Set default tags from the getTags function
+	defaultTags := getTags()
+	userTags := map[string]string{}
+
+	for _, param := range params {
+		parts := strings.SplitN(param, "=", 2)
+		if len(parts) == 2 {
+			if parts[0] == "Tags" {
+				tagEntries := strings.Split(parts[1], ",")
+				for _, entry := range tagEntries {
+					tagParts := strings.SplitN(entry, "=", 2)
+					if len(tagParts) == 2 {
+						userTags[tagParts[0]] = tagParts[1]
+					}
+				}
+			} else {
+				result[parts[0]] = parts[1]
+			}
+		}
+	}
+
+	// Combine default tags and user tags
+	combinedTags := mergeTags(defaultTags, userTags)
+
+	return result, combinedTags
+}
+
+// mergeTags combines default tags and user tags, giving precedence to user tags in case of conflicts
+func mergeTags(defaultTags, userTags map[string]string) map[string]string {
+	combinedTags := make(map[string]string)
+	for k, v := range defaultTags {
+		combinedTags[k] = v
+	}
+	for k, v := range userTags {
+		combinedTags[k] = v
+	}
+	return combinedTags
+}
+
+// selectTemplate selects the appropriate template file based on the parameters
+func selectTemplate(command string) string {
+	templates := map[string]string{
+		"rosa-quickstart-default-vpc":     "cmd/bootstrap/templates/vpc.yaml",
+		"rosa-quickstart-zero-egress-vpc": "",
+		// Add other templates here if needed
+	}
+
+	return templates[command]
+}
+
+// createVPCStack creates a CloudFormation stack to deploy a VPC
+func createVPCStack(templateFile string, params map[string]string, tags map[string]string) error {
+	// Load the AWS configuration
+	logger := logrus.New()
+	logger.SetLevel(logrus.DebugLevel)
+	cfg, err := config.LoadDefaultConfig(context.TODO(), config.WithRegion(params["Region"]))
+	if err != nil {
+		return fmt.Errorf("unable to load SDK config, %v", err)
+	}
+
+	// Read the CloudFormation template
+	templateBody, err := os.ReadFile(templateFile)
+	if err != nil {
+		return fmt.Errorf("unable to read template file, %v", err)
+	}
+
+	var cfTags []cfTypes.Tag
+	for k, v := range tags {
+		cfTags = append(cfTags, cfTypes.Tag{
+			Key:   aws.String(k),
+			Value: aws.String(v),
+		})
+	}
+
+	// Create a CloudFormation client
+	logger.Info("Creating CloudFormation client")
+	cfClient := cloudformation.NewFromConfig(cfg)
+
+	// Create the stack
+	logger.Info("Creating CloudFormation stack")
+	_, err = cfClient.CreateStack(context.TODO(), &cloudformation.CreateStackInput{
+		StackName:    aws.String(params["Name"]),
+		TemplateBody: aws.String(string(templateBody)),
+		Parameters: []cfTypes.Parameter{
```

Comment:
@den-rgb Again I'd expect this part to be generic in that we simply convert your `map[string]string` into a `[]cfTypes.Parameter`. So something like:

```
var cfParams []cfTypes.Parameter

for k,v := range params {
  cfParams = append(cfParams, {"ParameterKey": aws.String(k), "ParameterValue": aws.String(v)}
}
```